app/controllers/consent_forms_controller.rb

```diff
@@ -1,30 +1,44 @@
 class ConsentFormsController < ApplicationController
-  before_action :set_consent_form, only: [:show]
-  before_action :set_school, only: [:show]
-  before_action :set_session, only: [:unmatched_responses]
-
-  def show
-  end
-
   def unmatched_responses
+    @session =
+      policy_scope(Session).find(
+        params.fetch(:session_id) { params.fetch(:id) }
+      )
     @unmatched_consent_responses =
       @session.consent_forms.unmatched.recorded.order(:recorded_at)
   end
 
-  private
-
-  def set_consent_form
-    @consent_form = ConsentForm.find(params[:id])
+  def show
+    @consent_form = policy_scope(ConsentForm).find(params[:id])
+    @patient_sessions =
+      @consent_form
+        .session
+        .patient_sessions
+        .includes(:patient)
+        .order("patients.last_name")
   end
 
-  def set_school
-    @school = @consent_form.session.location
+  def review_match
+    @consent_form = policy_scope(ConsentForm).find(params[:id])
+    @patient_session =
+      policy_scope(PatientSession).find(params[:patient_session_id])
   end
 
-  def set_session
-    @session =
-      policy_scope(Session).find(
-        params.fetch(:session_id) { params.fetch(:id) }
-      )
+  def match
+    @consent_form = policy_scope(ConsentForm).find(params[:id])
+    @patient_session =
+      policy_scope(PatientSession).find(params[:patient_session_id])
+
+    Consent.from_consent_form(@consent_form, @patient_session)
```

Comment:
`from_consent_form` was built in a pretty rushed way. I think I'd like to amend it to either be a `!` method as it introduces a couple of side effects, or probably more flexibly, it should return a consent that you can then `.save` if you wish. Also the state machine stuff in it should probably be taken out and left to the caller.

Reply:
I've added a bang in [d32d318](https://github.com/nhsuk/manage-childrens-vaccinations/pull/1040/commits/d32d318166b42b695866a11ca175539a5c73d949). As discussed, we can roll forward and improve it when we do more work on it.